docs/DafnyRef/Specifications.md

```diff
@@ -503,6 +529,11 @@ a function from `int` to `set<object>`, then `reads F` has the meaning
 set x: int, o: object | o in F(x) :: o
 ```
 
+For each function value `f`, Dafny defines the function `f.reads`,
+which takes the same arguments as `f` and returns that set of objects
+that `f` reads (according to its reads clause) with those arguments.
+`f.reads` has type `int ~> set<object>`.
```

Comment:
Are you referring to a particular function `f` here?

Reply:
No, its 'forall' functions f

Reply:
Then, why do you write `f.reads` has type `int ~> set<object>`. Shouldn't it be, say, `T` instead of `int`?

Reply:
f is a function,. f.reads is also a function. It takes the same arguments as f, but returns the set of objects read. Similarly f.requires takes the same arguments as f but returns a boolean (wither the precondition is satisfied or not).

The use case is to be able to include, say f.requires in a precondition of another funciton/method that has f as an argument.  So map(f: T->bool, s: seq<T>): seq<bool> might have forall t: t in s :: f.requires(t) as a precondition.

Reply:
Change the int to T